stores/metadata.go

```diff
@@ -397,6 +410,89 @@ func (raw rawObject) toSlabSlice() (slice object.SlabSlice, _ error) {
 	return slice, nil
 }
 
+func (s *SQLStore) Bucket(_ context.Context, bucket string) (api.Bucket, error) {
+	var b dbBucket
+	err := s.db.
+		Model(&dbBucket{}).
+		Where("name = ?", bucket).
+		Take(&b).
+		Error
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		return api.Bucket{}, api.ErrBucketNotFound
+	} else if err != nil {
+		return api.Bucket{}, err
+	}
+	return api.Bucket{
+		CreationDate: b.CreatedAt.UTC(),
+		Name:         b.Name,
+	}, nil
+}
+
+func (s *SQLStore) CreateBucket(_ context.Context, bucket string) error {
+	// Create bucket.
+	return s.retryTransaction(func(tx *gorm.DB) error {
+		res := tx.Clauses(clause.OnConflict{
+			DoNothing: true,
+		}).
+			Create(&dbBucket{Name: bucket})
+		if res.Error != nil {
+			return res.Error
+		} else if res.RowsAffected == 0 {
+			return api.ErrBucketExists
```

Comment:
is this what we want? we don't check for it and it's a PUT route

Reply:
Changed it to a `POST` since we want this behaviour.